### PR TITLE
feat: imagebuilder for envoy and squid

### PR DESCRIPTION
**Pull request by dracarys18**

### CODE REVIEW

File: lib/aws/image_builder_stack.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            // env: {
+            //   account: process.env.CDK_DEFAULT_ACCOUNT,
+            //   region: process.env.CDK_DEFAULT_REGION
+            // },
+            stackName: config.name,
+        });
+
+        let vpcConfig: VpcConfig = {
+            name: "imagebuilder-vpc",
+            availabilityZones: [process.env.CDK_DEFAULT_REGION + "a", process.env.CDK_DEFAULT_REGION + "b"]
```
[the remark this answers is not in the document]
Comment (dracarys18) [reply lib/aws/image_builder_stack.ts:25]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html. From this document every region has multiple availability zone.

File: deploy_imagebuilder.sh [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    done
+    printf "\r%s [Done]   \n" "$message"
+}
+
+# Function to show installation options
+show_install_options() {
+    echo
+    echo "${bold}Choose the base image type:${reset}"
+    echo "${bold}${blue}1.${reset}${green}Hardened base Amazon Linux ${reset}"
+    echo -e "${bold}${blue}2.${reset}${green}Normal Amazon Linux Image${reset}\n"
```
Comment (Niteshballa) [inline deploy_imagebuilder.sh:40]: Can we use printf here? In POSIX sh, echo flags are undefined.

File: deploy_imagebuilder.sh [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    echo "AWS CLI could not be found. Please rerun 'bash install.sh' with Sudo access and ensure the command is available within the $PATH"
+    exit 1
+fi
+
+echo "Dependency installation completed."
+
+echo
+echo "${blue}##########################################${reset}"
+echo "${blue}    Checking neccessary permissions${reset}"
+echo -e "${blue}##########################################${reset}\n"
```
Comment (Niteshballa) [inline deploy_imagebuilder.sh:172]: Same here

**Comment by Niteshballa** [comment]
Do we need to support single click deployment of image builder as well?

**Comment by nitesh-balla** [comment]
@dracarys18 Should we add about image builder and deploying it in the readme file?

**Pull request merged**